cves/spookyssl.py

```diff
@@ -0,0 +1,252 @@
+                    else:
+                        state[pid] = status.process_not_determined(pid, vulnerability)
+    return state
+
+
+def vector_three(state, debug, container_name):
+    """This function performs the "vector three" of checking exploitability which is checking if there is a running
+    process that loads libcrypto/libssl/openssl files that are using an affected OpenSSL version."""
+    vulnerability = f'{VULNERABILITY} (the so files check)'
+    pids = process_functions.running_processes(debug, container_name)
+    if pids:
+        print(constants.FULL_QUESTION_MESSAGE.format('Are there running processes that load "openssl/libssl/libcrypto" '
+                                                     'so files??'))
+        process_state = validate_processes_vector_three(pids, vulnerability, debug, container_name)
+        if process_state:
+            status.remediation_mitigation(REMEDIATION, MITIGATION)
+            state[vulnerability] = process_state
+        else:
+            print(constants.FULL_POSITIVE_RESULT_MESSAGE.format('No'))
+            print(constants.FULL_EXPLANATION_MESSAGE.format(f'There are no running processes loading so files that have'
+                                                            f' an affected OpenSSL version'))
+            state[vulnerability] = status.not_vulnerable(vulnerability)
+    else:
+        state[vulnerability] = status.not_vulnerable(vulnerability)
+    return state
+
+
+def validate_processes_vector_two(state, pids, vulnerability, debug, container_name):
+    """This function loops over all Node processes and checks if their versions known to be using an affected OpenSSL
+    version."""
+    for pid in pids:
+        print(constants.FULL_QUESTION_MESSAGE.format('Is Node version affected?'))
+        version_output = process_functions.process_executable_version(pid, debug, container_name)
+        if version_output == constants.UNSUPPORTED:
+            state[pid] = status.process_not_determined(pid, vulnerability)
+        node_version = version_output[constants.FIRST:]
+        if version.parse(node_version) >= version.parse(MIN_VULNERABLE_NODE_VERSIONS):
+            print(constants.FULL_NEGATIVE_RESULT_MESSAGE.format('Yes'))
+            print(constants.FULL_EXPLANATION_MESSAGE.format(f'Node versions that are equal or higher than 17.0.0 are '
+                                                            f'using OpenSSL versions 3.0.x\nYour node version which is:'
+                                                            f' {node_version}, is affected'))
+            state[pid] = status.process_vulnerable(pid, vulnerability)
+            status.remediation_mitigation(REMEDIATION, MITIGATION)
+        else:
+            print(constants.FULL_POSITIVE_RESULT_MESSAGE.format('No'))
+            print(constants.FULL_EXPLANATION_MESSAGE.format(f'Node versions that are equal or higher than 17.0.0 are '
```

Comment:
We need to rephrase this comment. As it is now, it wouldn't be true once patches will be released.